test/e2e/v3/plugin_cluster_test.go

```diff
@@ -214,6 +214,16 @@ func Run(kbc *utils.TestContext) {
 		fmt.Sprintf("--serviceaccount=%s:%s", kbc.Kubectl.Namespace, kbc.Kubectl.ServiceAccount))
 	ExpectWithOffset(1, err).NotTo(HaveOccurred())
 
+	// As of Kubernetes 1.24 a ServiceAccount no longer has a ServiceAccount token
+	// secret autogenerated. We have to create it manually here
+	By("Creating the ServiceAccount token")
+	secretFile, err := createServiceAccountToken(kbc.Kubectl.ServiceAccount, kbc.Dir)
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", secretFile)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
```

Comment:
We will need to call that on the tests for we not use it here: https://github.com/kubernetes-sigs/kubebuilder/blob/af78fcc8deee45913809303709b902a81fc0bfbe/test/e2e/v3/plugin_cluster_test.go#L87-L95

Also, we need to do in the createServiceAccountToken and as a follow up we need to check if we should not be scaffolding it.